components/camel-sql/src/main/java/org/apache/camel/component/sql/stored/template/ast/ParseHelper.java

```diff
@@ -21,29 +21,75 @@
 
 import org.apache.camel.component.sql.stored.template.generated.SSPTParserConstants;
 import org.apache.camel.component.sql.stored.template.generated.Token;
+import org.springframework.util.ClassUtils;
 import org.springframework.util.ReflectionUtils;
 
 public final class ParseHelper {
 
     private ParseHelper() {
     }
 
-    public static int parseSqlType(Token sqlType) {
+    public static int parseSqlType(Token sqlTypeToken) {
+
+        String sqlType = sqlTypeToken.toString();
 
         //if number then use it(probably Vendor spesific SQL-type)
-        if (sqlType.kind == SSPTParserConstants.NUMBER) {
-            return Integer.valueOf(sqlType.toString());
+        if (sqlTypeToken.kind == SSPTParserConstants.NUMBER) {
+            return Integer.valueOf(sqlType);
+        }
+
+        //if contains .
+        if (sqlType.contains(".")) {
+            String className;
+            String fieldName;
+            try {
+                className = sqlType.substring(0, sqlType.lastIndexOf("."));
+                fieldName = sqlType.substring(sqlType.lastIndexOf(".") + 1);
+            } catch (Exception ex) {
+                throw new ParseRuntimeException("Failed to parse class.field:" + sqlType);
+            }
+            try {
+                Class clazz = ClassUtils.forName(className, ParseHelper.class.getClassLoader());
```

Comment:
There is a ClassResolver via CamelContext you can get that we need to use instead to load classes. This ensures class loading works on different runtimes. This may require to add CamelContext as a parameter to this method, so you can use it.